lesson31/src/js/drawer-menu.js

```diff
@@ -8,5 +7,47 @@ const toggleDrawerMenu = () => {
   body.classList.toggle("drawer-menu-open");
 }
 
+const createOverLay = () => {
+  const overLay = document.createElement("div");
+  overLay.id = "js-drawer-overlay";
+  overLay.classList.add("drawer-overlay");
+  return overLay;
+}
+
+const renderOverLay = () => {
+  const overLay = createOverLay();
+  overLay.addEventListener("click", toggleDrawerMenu);
+  body.appendChild(overLay);
+}
+
+const switchDrawerDirect = (option) => {
+  switch (option) {
+    case "left":
+      {
+        drawerMenu.classList.add("drawer--left");
+        break;
+      }
+    case "right":
+      {
+        drawerMenu.classList.add("drawer--right");
+        break;
+      }
+    default:
+      drawerMenu.classList.add("drawer--left");
+  }
+}
+
+const settingDrawerMenu = (options) => {
+  const drawerOptions = {
+    direct: "left",
+    overLay: true,
+    duration: 0.3,
+    ...options
+  }
+  drawerOptions.overLay && renderOverLay();
+  switchDrawerDirect(drawerOptions.direct);
+  drawerMenu.style.transitionDuration = `${drawerOptions.duration}s`;
+}
```

Comment:
optionなので引数がなくても動くようにしたいです

```suggestion
const settingDrawerMenu = (options = {}) => {
  const defaultOption = { direct: "right", overLay: true, duration: 0.3 };
  const drawerOptions = {
    ...defaultOption,
    ...options
  };
  drawerOptions.overLay && renderOverLay();
  switchDrawerDirect(drawerOptions.direct);
  drawerMenu.style.transitionDuration = `${drawerOptions.duration}s`;
};
settingDrawerMenu();
```

Reply:
Thank you.The empty case was not considered.
Fixed.
- Add default option variable -> [653f541](https://github.com/sae-github/JS_lesson/pull/39/commits/653f541d301bf8bd949cbf248a60bf6f567df230)
- Set default values -> [1b48266](https://github.com/sae-github/JS_lesson/pull/39/commits/1b48266d34296914ccdbbf1719f0bf9a38c2cb4f)